Documentation/Books/Manual/Programs/Arangodump/Maskings.md

```diff
@@ -88,14 +109,140 @@ or
 
     "path": "`name.with.dots`"
 
-xify_front
-----------
+If the attribute value is an array the masking is applied to all the
+array elements.
+
+### Example
+
+This following will replace name by a "XXXX" masked string
+
+    {
+      "type": "xify_front",
+      "path": ".name",
+      "unmaskedLength": 2
+    }
+
+The object
+
+    {
+      "name": "top-level-name",
+      "age": 42,
+      "nicknames" : [ { "name": "hugo" }, "egon" ],
+      "other": {
+        "name": [ "emil", { "secret": "superman" } ]
+      }
+    }
+
+will be replaced as follows:
+
+    {
+      "name": "xxxxxxxxxxxxme",
+      "age": 42,
+      "nicknames" : [ { "name": "xxgo" }, "egon" ],
+      "other": {
+        "name": [ "xxil", { "secret": "superman" } ]
+      }
+      }
+
+`egon` and `superman` are not replaced, because they are not in an
+attribute value of an attribute `name`.
+
+### sub-objects and arrays
+
+If you specify a path and the attribute value is an array then the
+masking decision is applied to each entry of this array as if this
+were the value of the attribute.
+
+If the attribute value is a object, then the attribute is not masked.
+Instead the sub-object is checked further for leaf attributes.
+
+#### Example
+
+Masking `email` will convert
 
-This masking replaces characters with `x` and ` `. Alphanumeric characters,
-`_` and `-` are replaced by `x`, everything else is replaced by ` `.
+
+  { 
+    "email" : "email address" 
+  }
+
+into
+
+  { 
+    "email" : "xxil xxxxxxss" 
+  }
+
+because `email` is a leaf sttribute.
+
+  { 
+    "email" : [ 
+      "address one", 
+      "address two" 
+    ] 
+  } 
+
+will be converted into
+
+  { 
+    "email" : [ 
+      "xxxxxss xne", 
+      "xxxxxss xwo" 
+    ] 
+  } 
+
+because the array is "unfolded".
+
+  { 
+    "email" : { 
+      "address" : "email address" 
+    } 
+    }
+
+will not be changed because `email` is not a leaf attribute.
+
+
+Masking Functions
+-----------------
+
+### random_string
+
+This masking replaces character strings with a random string of the
```

Comment:
I assume it applies to all kinds of strings? _Character strings_ makes it sound a bit like it would only replace strings with letters, but not number string (_key) for instance.

Reply:
done